src/dbnode/storage/shard.go

```diff
@@ -1017,6 +1028,12 @@ func (s *dbShard) SeriesReadWriteRef(
 		}, nil
 	}
 
+	// BEFORE MERGE: CONSIDER BELOW FOR NOW OR FOLLOWUP
+	// TODO(r): Probably can't insert series sync otherwise we stall a ton
+	// of writes... need a better solution for bootstrapping.
+	// This is what causes writes to degrade during bootstrap.
+	// This is a note to consider before merging this PR.
```

Comment:
Discussed over call, seems like we can tackle this after this PR gets merged.

Reply:
Agreed will remove the comment about solving this before merging PR.